From be2e1a62fefe03ac50dc0e0594af4944b07319e0 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Mon, 6 Nov 2017 17:24:53 +0000
Subject: [PATCH] libedit: check for null before passing cmdedit_prompt to
 fputs

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
Upstream-Status: Pending

 libbb/lineedit.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libbb/lineedit.c b/libbb/lineedit.c
index 543a3f11c..5ecf9dd76 100644
--- a/libbb/lineedit.c
+++ b/libbb/lineedit.c
@@ -505,7 +505,9 @@ static void put_prompt_custom(bool is_full)
 	/* https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html
 	 * says that shells must write $PSn to stderr, not stdout.
 	 */
-	fputs((is_full ? cmdedit_prompt : prompt_last_line), stderr);
+	if ( !is_full || NULL != cmdedit_prompt ) {
+		fputs((is_full ? cmdedit_prompt : prompt_last_line), stderr);
+	}
 	cursor = 0;
 	cmdedit_y = cmdedit_prmt_len / cmdedit_termw; /* new quasireal y */
 	cmdedit_x = cmdedit_prmt_len % cmdedit_termw;
